tumblr/tumblr.php

```diff
@@ -247,11 +247,13 @@ function tumblr_settings(App $a, &$s)
 
 	$s .= '<div id="tumblr-enable-wrapper">';
 	$s .= '<label id="tumblr-enable-label" for="tumblr-checkbox">' . L10n::t('Enable Tumblr Post Addon') . '</label>';
+	$s .= '<input type="hidden" name="tumblr" value="0"/>';
 	$s .= '<input id="tumblr-checkbox" type="checkbox" name="tumblr" value="1" ' . $checked . '/>';
 	$s .= '</div><div class="clear"></div>';
 
 	$s .= '<div id="tumblr-bydefault-wrapper">';
 	$s .= '<label id="tumblr-bydefault-label" for="tumblr-bydefault">' . L10n::t('Post to Tumblr by default') . '</label>';
+	$s .= '<input type="hidden" name="tumblr_bydefault" value="0"/>';
 	$s .= '<input id="tumblr-bydefault" type="checkbox" name="tumblr_bydefault" value="1" ' . $def_checked . '/>';
```

Comment:
Out of curiosity, why does adding the `tumblr_bydefault` field a 2nd time with a different value fix the problem?

Reply:
Checkboxes have this curious property that if they aren't checked, the associated form field is considered not to exist for the form data purpose. Additionally, when two form fields exist with the same name (excluding array names like `name[]`), only the latter value is used for the field. You can combine both effects to ensure a checkbox form field always has a value for its name:
- When the checkbox is unchecked, the only form field present with the name is the hidden field whose value gets used.
- When the checkbox is checked, its value appears in the form later than the hidden field and overrides it in the sent form data.

Reply:
Thanks!

Do you know when this behavior changed? Roughly in browser generations, because I don't remember this quirk being there in the *good old days*--I maybe I just never encountered the behavior as bug x)

Reply:
I have always known checkboxes to work like this and the field not to be included at all in submitted form data when the checkbox is unchecked. There's never been a value attribute for the unchecked state, only for the checked one.